providers/catalog.go

```diff
@@ -42,6 +42,7 @@ const (
 	Slack                               Provider = "slack"
 	HelpScoutMailbox                    Provider = "helpScoutMailbox"
 	Webflow                             Provider = "webflow"
+	GMail                               Provider = "gmail"
```

Comment:
I know that this is technically correct, but on second thought, let's change this to `GoogleMail` so that we can use it as a convention for `GoogleContacts` or just `Google` for the base connector?